adapters/amqp-vertx-quarkus/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -1198,6 +1196,8 @@ protected void onCommandReceived(
             items.put(TracingHelper.TAG_QOS.getKey(), sender.getQoS().name());
             items.put(TracingHelper.TAG_CREDIT.getKey(), sender.getCredit());
             commandContext.getTracingSpan().log(items);
+            // return succeeded future, not waiting for the delivery update from the device
+            return Future.succeededFuture();
```

Comment:
so, the only reason why we return a Future in the first place, is to be able to throttle the consumption of messages from Kafka based on when the Future gets completed, right?

Reply:
Yes. The throttling mechanism is limiting the number of records being "in processing" at any given point in time. The "in processing" state here refers to the time between invoking the record handler and the result Future getting completed.

Reply:
got it